packages/components/src/components/post-linkarea/post-linkarea.tsx

```diff
@@ -1,33 +1,53 @@
-import { Component, Element, h, Host } from '@stencil/core';
+import { Component, Element, h, Host, State } from '@stencil/core';
 import { version } from '@root/package.json';
 
 const INTERACTIVE_ELEMENTS = ['a'].join(',');
 const INTERACTIVE_ELEMENTS_SELECTOR = `:where(${INTERACTIVE_ELEMENTS})`;
 
-type InteractiveElement = HTMLAnchorElement;
-
 @Component({
   tag: 'post-linkarea',
   styleUrl: 'post-linkarea.scss',
   shadow: true,
 })
 export class PostLinkarea {
+  private mutationObserver = new MutationObserver(this.checkInteractiveElements.bind(this));
+
   @Element() host: HTMLPostLinkareaElement;
 
+  @State() interactiveElements: NodeListOf<HTMLAnchorElement>;
+
   private dispatchClick({ ctrlKey, shiftKey, altKey, metaKey }: MouseEvent) {
-    const interactiveElement: InteractiveElement =
-      this.host.querySelector(`[data-link]${INTERACTIVE_ELEMENTS_SELECTOR}`) ??
-      this.host.querySelector(INTERACTIVE_ELEMENTS_SELECTOR);
+    this.interactiveElements[0].dispatchEvent(
+      new MouseEvent('click', { ctrlKey, shiftKey, altKey, metaKey }),
+    );
+  }
+
+  private checkInteractiveElements() {
+    this.interactiveElements = this.host.querySelectorAll(INTERACTIVE_ELEMENTS_SELECTOR);
 
-    if (!interactiveElement) {
+    if (!this.interactiveElements.length) {
       throw new Error(
         `The \`post-linkarea\` component must contain an interactive element. Possible elements are: ${INTERACTIVE_ELEMENTS}.`,
       );
     }
```

Comment:
Since we don't know if the link is not there because there is simply no link or because it will be added asynchronously later, I'd remove this error altogether: 
```suggestion
```